test/query_test/query_test.go

```diff
@@ -50,6 +51,81 @@ var _ = Describe("Service Manager Query", func() {
 		}
 	})
 
+	Context("Service instance and last operations query test", func() {
+		var serviceInstance1, serviceInstance2 *types.ServiceInstance
+		BeforeEach(func() {
+			_, serviceInstance1 = common.CreateInstanceInPlatform(ctx, ctx.TestPlatform.ID)
+			_, serviceInstance2 = common.CreateInstanceInPlatform(ctx, ctx.TestPlatform.ID)
+		})
+
+		AfterEach(func() {
+			ctx.CleanupAdditionalResources()
+		})
+
+		It("should return the last operation for a newly created resource", func() {
+			queryParams := map[string]interface{}{
+				"id_list": []string{serviceInstance1.ID},
+			}
+
+			list, err := repository.QueryForList(context.Background(), types.OperationType, storage.QueryForLastOperationsPerResource, queryParams)
+			Expect(err).ShouldNot(HaveOccurred())
+			lastOperation := list.ItemAt(0).(*types.Operation)
+			Expect(err).ShouldNot(HaveOccurred())
+			Expect(list.Len()).To(BeEquivalentTo(1))
+			Expect(lastOperation.State).To(Equal(types.SUCCEEDED))
```

Comment:
is this enough to be sure that this is the last one? don't you want to compare IDs?

Reply:
i am still not sure that this is the right operation.. please don't resolve comments without making any change or leaving a reply

Reply:
> i am still not sure that this is the right operation.. please don't resolve comments without making any change or leaving a reply

tests was extended with both:
Expect(lastOperation.State).To(Equal(types.SUCCEEDED))
Expect(lastOperation.Type).To(Equal(types.CREATE))

as we create the instance, we know that it should have exactly one created SUCCEEDED operation

Reply:
the other resource also has a single SUCCEEDED & created operation. you query for a specific resource, it makes sense to verify that you got the operation for that resource and not another. just add `Expect(lastOperation.ResourceID).To(Equal(serviceInstance1.ID))`.

Reply:
Ahh got it .. i thought you wanted to make sure that it returns that correct "last operation" of the instance in the case of multiple ops for the same resource. I will add an expect to confirm the instance id.. got it 👍